Community Toolbox/scripts/utils_CommunityToolboxDraw/utils_CommunityToolboxDraw.gml

```diff
@@ -0,0 +1,266 @@
+#region Settings
+
+/// @func draw_set_color_alpha(color,alpha)
+/// @desc Sets the drawing color and alpha.
+/// @arg {Constant.Color} color     The drawing color to set.
+/// @arg {Real} alpha               The alpha to set (between 0 and 1).
+function draw_set_color_alpha(_color, _alpha) {
+    gml_pragma("forceinline");
+    draw_set_color(_color);
+    draw_set_alpha(_alpha);
+}
+
+/// @func draw_set_colour_alpha(colour,alpha)
+/// @desc Sets the drawing colour and alpha.
+/// @arg {Constant.Color} colour    The drawing colour to set.
+/// @arg {Real} alpha               The alpha to set (between 0 and 1).
+function draw_set_colour_alpha(_colour, _alpha) {
+    gml_pragma("forceinline");
+    draw_set_colour(_colour);
+    draw_set_alpha(_alpha);
+}
+
+/// @func draw_set_align(halign,valign)
+/// @desc Sets the horizontal and vertical alignment of the drawn text.
+/// @arg {Constant.HAlign} halign       The horizontal alignment to set.
+/// @arg {Constant.VAlign} valign       The vertical alignment to set.
+function draw_set_align(_halign, _valign) {
+    gml_pragma("forceinline");
+    draw_set_halign(_halign);
+    draw_set_valign(_valign);
+}
+
+#endregion
+
+#region Shapes
+
+/// @func draw_arc(x,y,radius,anglefrom,angleto,[precision])
+/// @desc Draws a circular line around the given center, with the given radius and between given angles.
+/// @arg {Real} x               The x coordinate of the arc center.
+/// @arg {Real} y               The y coordinate of the arc center.
+/// @arg {Real} radius          The radius of the arc around the center.
+/// @arg {Real} anglefrom       The starting angle of the arc.
+/// @arg {Real} angleto         The ending angle of the arc.
+/// @arg {Real} [precision]     The precision of the circle the arc is a part of.
+function draw_arc(_x, _y, _radius, _anglefrom, _angleto, _precision = 24) {
+    // static functions    
+    static get_segment_position = function(_angle, _segment_angle) {
```

Comment:
I've stared at this helper function for ten minutes and I can't figure out what the math is doing here. Not a criticism, but I can't review this part since I have no idea what it's trying to do :cold_sweat:

Reply:
This comment might be relevant here: https://github.com/Alphish/gm-community-toolbox/issues/5#issuecomment-2296658951

The approach here is to "cut out" an arc from a "circle" which is actually some 24-agon or 8-agon or 360-agon or whatever precision will be given here. So e.g. a 24-agon circle would have 24 segments, but then I'd need to calculate where the starting point lands on the initial segment.

So `get_segment_position` answers the question of: "if I have a segment spanning [segment angle] degrees and the initial point being [angle] degrees through the segment, how large portion of the segment length has the initial point passed".

E.g. for the segment angle of 20 degrees, the initial point angle of 0 degrees would be 0% way through, 10 degrees would be 50% way through and 20 degrees would be 100% way through. But because angles are tricky, 5 degrees won't necessarily give 25% way through but some other number, and that's the number I'm trying to find out here.

Reply:
Yep, I think I got it now! :+1: